app/Http/Controllers/Api/SocialAssistant/SocialAssistantSubjectController.php

```diff
@@ -0,0 +1,638 @@
+                'type' => 'success',
+                'message' => __('messages.common.success_view'),
+                'data' => SubjectResource::make($subject),
+            ], HttpResponse::HTTP_OK);
+        } catch (\Exception $e) {
+            return response()->json(['type' => 'error', 'message' => $e->getMessage()], HttpResponse::HTTP_BAD_REQUEST);
+        }
+    }
+
+    /**
+     * @OA\Get(
+     *     path="/api/social-assistant/get/form-infos",
+     *     operationId="SocialAssistantGetSubjectFormInfos",
+     *     tags={"SocialAssistant/Subject"},
+     *     summary="Get subject form infos",
+     *     description="Get subject form infos",
+     *     security={{"sanctum":{}}},
+     *
+     *     @OA\Response(
+     *         response=200,
+     *         description="Get data successfully",
+     *
+     *         @OA\JsonContent(
+     *
+     *               @OA\Property(
+     *                    property="skinColors",
+     *                    type="array",
+     *
+     *                    @OA\Items(
+     *                        type="object",
+     *
+     *                        @OA\Property(property="id", type="string", example="black"),
+     *                        @OA\Property(property="name", type="string", example="Preto(a)"),
+     *                    )
+     *                ),
+     *                @OA\Property(
+     *                     property="states",
+     *                     type="array",
+     *
+     *                     @OA\Items(
+     *                         type="object",
+     *
+     *                         @OA\Property(property="id", type="integer", example="1"),
+     *                         @OA\Property(property="name", type="string", example="Acre"),
+     *                     )
+     *                 )
+     *         )
+     *     ),
+     *
+     *     @OA\Response(
+     *         response=400,
+     *         description="Bad Request",
+     *
+     *         @OA\JsonContent(
+     *
+     *             @OA\Property(property="type", type="string", example="error"),
+     *             @OA\Property(property="message", type="string", example="Bad Request")
+     *         )
+     *     ),
+     *
+     *     @OA\Response(
+     *         response=403,
+     *         description="Forbidden",
+     *
+     *         @OA\JsonContent(
+     *
+     *             @OA\Property(property="type", type="string", example="error"),
+     *             @OA\Property(property="message", type="string", example="This action is unauthorized.")
+     *         )
+     *     ),
+     * )
+     */
+    public function getFormInfos(): JsonResponse
+    {
+        try {
+            $skinColors = collect(SkinColorsEnum::cases())
+                ->map(function ($skinColor) {
+                    return ['id' => $skinColor->value, 'name' => SkinColorsEnum::trans($skinColor->value)];
+                });
+
+            $states = to_select(State::all());
```

Comment:
Certeza que isso fica no controller da assistente social e subject?

Eu separaria as rotas de obter estados e cidades em um controller específico, talvez StateCityController, ou algo assim. Se essa informação for útil em outro lugar, temos a rota pronta. As rotas seriam genéricas também, não estando dentro do escopo da assistente;

Reply:
Se você relamente só usa essas informações aqui no controller de criação de sujeitos, então ok deixar assim;